ecr/struct.ecr

```diff
@@ -20,8 +24,7 @@ module <%= namespace_name %>
                    end
       <% elsif object.copyable? %>
         # Raw structs are always moved to Crystal memory.
-        @pointer = Pointer(Void).malloc(sizeof(<%= to_lib_type(object) %>))
-        @pointer.copy_from(pointer, sizeof(<%= to_lib_type(object) %>))
+        @data = pointer.as(Pointer(<%= to_lib_type(object) %>)).value
```

Comment:
What if the struct have a C string? the copy will have a pointer to a invalid memory, freed by the free call above..

I think we must do this optimization just for structs without pointers, and do like you said... using Crystal structs instead of class, then things like `Point`, etc... will work as fast as in C.

So we probably need a `StructInfo#pod_type?` or `StructInfo#trivial_type?` to return true if the struct have only non-pointers as field members.

Reply:
It's true!